winch/codegen/src/codegen/env.rs

```diff
@@ -0,0 +1,19 @@
+use wasmparser::FuncType;
+
+/// Function environment used the by the code generation to
+/// resolve module and runtime-specific information.
+pub trait FuncEnv {
+    /// Get the callee information from a given function index.
+    fn callee_from_index(&self, index: u32) -> Callee;
+}
+
+/// Metadata about a function callee.  Use by the code generation
+/// to emit function calls.
+pub struct Callee {
+    /// The function type.
+    pub ty: FuncType,
+    /// A flag to determine if the callee is imported.
+    pub import: bool,
+    /// The callee index.
```

Comment:
Is this the same as the `index` that is passed into `callee_from_index`? Can we document that if so (something like "The callee index in the Wasm function index space")?

Reply:
Agreed and yeah it's the same index -- I'm returning it as part of the `Callee` for convenience. 

Fixed!